theano/sandbox/cuda/dnn.py

```diff
@@ -2597,7 +2597,8 @@ def local_softmax_dnn_grad(node):
                 gpu_contiguous(ins[0]),
```

Comment:
I don't think that is correct. This GpuDnnSoftmaxGrad expects the softmax to have been computed on the 'channel' axis, and expects 'bc01' as a layout. Now, that axis is of length 1.
If we do these dimshuffles, then we would need to pass 'instance' instead of 'channel', I think.